obs2aptly/src/lib.rs

```diff
@@ -659,6 +753,85 @@ impl SyncActions {
         &self.actions
     }
 
+    #[tracing::instrument(skip_all)]
+    async fn query_pool_packages(&self) -> Result<PoolPackagesByName> {
+        // Querying for all the packages results in a URL that is far too long
+        // (over the 65k limit set by reqwest), so split it into 1k packages per
+        // query.
+        const CHUNK_SIZE: usize = 1000;
+
+        let query_parts: Vec<_> = self
+            .actions
+            .iter()
+            .filter_map::<&str, _>(|a| match a {
+                SyncAction::AddDeb { package, .. } => Some(package),
+                SyncAction::AddDsc { package, .. } => Some(package),
+                _ => None,
+            })
+            .collect();
+
+        let mut packages: Vec<packages::Package> = vec![];
+        for chunk in query_parts.chunks(CHUNK_SIZE) {
+            let query = chunk.to_vec().join("|");
```

Comment:
why do you need to_vec here?

Reply:
oops, I think this was an iterator before :sweat_smile: